src/components/Audiences/index.js

```diff
@@ -18,12 +18,17 @@ const createAudiences = ({ logger }) => {
     fireReferrerHideableImage,
     logger
   });
+  const handleResponse = ({ response }) => {
+    if (!response) {
+      return undefined;
+    }
+    const destinations = response.getPayloadsByType("activation:push");
+    return processDestinations(destinations);
+  };
   return {
     lifecycle: {
-      onResponse({ response }) {
-        const destinations = response.getPayloadsByType("activation:push");
-        return processDestinations(destinations);
-      }
+      onResponse: handleResponse,
+      onRequestFailure: handleResponse
```

Comment:
maybe we should rename `handleResponse` to something more generic that applies to both lyfecycles?